src/server/pfs/s3/multipart.go

```diff
@@ -280,17 +275,20 @@ func (c *controller) CompleteMultipart(r *http.Request, bucketName, key, uploadI
 			return nil, s2.EntityTooSmallError(r)
 		}
 
-		err = pc.CopyFile(c.repo, "master", srcPath, bucket.Repo, bucket.Commit, key, false)
-		if err != nil {
-			if errutil.IsWriteToOutputBranchError(err) {
-				return nil, writeToOutputBranchError(r)
-			}
+		if err := pc.CopyFile(c.repo, "master", srcPath, c.repo, "master", dstPath, false); err != nil {
```

Comment:
@PFedak why does writing to the random path not create additional commits?

Reply:
nevermind i just noticed this is in another repo

Reply:
It does, but only in the special `_s3gateway_multipart_` repo, rather than the destination. It might have made sense to use different branches or even different repos for separate uploads so that they could use some of the put file client methods rather than all creating new commits, but this does make it easy to not worry about concurrency.